vault/core_test.go

```diff
@@ -634,7 +634,7 @@ func TestCore_HandleRequest_NoClientToken(t *testing.T) {
 	}
 
 	ct := noop.Requests[0].ClientToken
-	if ct == "" || ct == root {
+	if ct != "" {
```

Comment:
Does this mean that this test has always been broken?

Reply:
Nope. It was fine before. Its just that router doesn't passthrough the token to backends --that don't require it-- anymore.